backends/simplego/dotgeneral_large.go

```diff
@@ -514,63 +764,95 @@ func buildDotGeneralKernel[T PODNumericConstraints](lhs, rhs, output *Buffer, bl
 			// Loop 4 rows at a time.
 			for rhsRow := 0; rhsRow < blockDim; rhsRow += 4 { // range blockDim { // loop over rhs rows:
 				lhsIdx := baseLhsIdx
-				contractingIdx := 0
-				sum0 := outputFlat[outputIdx]
-				sum1 := outputFlat[outputIdx+1]
-				sum2 := outputFlat[outputIdx+2]
-				sum3 := outputFlat[outputIdx+3]
-				// Loop unrolled 8 at a time.
-				for ; contractingIdx+7 < blockDim; contractingIdx += 8 {
-					rhsIdx1 := rhsIdx + blockDim
-					rhsIdx2 := rhsIdx + 2*blockDim
-					rhsIdx3 := rhsIdx + 3*blockDim
-					sum0 += lhsFlat[lhsIdx]*rhsFlat[rhsIdx] +
-						lhsFlat[lhsIdx+1]*rhsFlat[rhsIdx+1] +
-						lhsFlat[lhsIdx+2]*rhsFlat[rhsIdx+2] +
-						lhsFlat[lhsIdx+3]*rhsFlat[rhsIdx+3] +
-						lhsFlat[lhsIdx+4]*rhsFlat[rhsIdx+4] +
-						lhsFlat[lhsIdx+5]*rhsFlat[rhsIdx+5] +
-						lhsFlat[lhsIdx+6]*rhsFlat[rhsIdx+6] +
-						lhsFlat[lhsIdx+7]*rhsFlat[rhsIdx+7]
-					sum1 += lhsFlat[lhsIdx]*rhsFlat[rhsIdx1] +
-						lhsFlat[lhsIdx+1]*rhsFlat[rhsIdx1+1] +
-						lhsFlat[lhsIdx+2]*rhsFlat[rhsIdx1+2] +
-						lhsFlat[lhsIdx+3]*rhsFlat[rhsIdx1+3] +
-						lhsFlat[lhsIdx+4]*rhsFlat[rhsIdx1+4] +
-						lhsFlat[lhsIdx+5]*rhsFlat[rhsIdx1+5] +
-						lhsFlat[lhsIdx+6]*rhsFlat[rhsIdx1+6] +
-						lhsFlat[lhsIdx+7]*rhsFlat[rhsIdx1+7]
-					sum2 += lhsFlat[lhsIdx]*rhsFlat[rhsIdx2] +
-						lhsFlat[lhsIdx+1]*rhsFlat[rhsIdx2+1] +
-						lhsFlat[lhsIdx+2]*rhsFlat[rhsIdx2+2] +
-						lhsFlat[lhsIdx+3]*rhsFlat[rhsIdx2+3] +
-						lhsFlat[lhsIdx+4]*rhsFlat[rhsIdx2+4] +
-						lhsFlat[lhsIdx+5]*rhsFlat[rhsIdx2+5] +
-						lhsFlat[lhsIdx+6]*rhsFlat[rhsIdx2+6] +
-						lhsFlat[lhsIdx+7]*rhsFlat[rhsIdx2+7]
-					sum3 += lhsFlat[lhsIdx]*rhsFlat[rhsIdx3] +
-						lhsFlat[lhsIdx+1]*rhsFlat[rhsIdx3+1] +
-						lhsFlat[lhsIdx+2]*rhsFlat[rhsIdx3+2] +
-						lhsFlat[lhsIdx+3]*rhsFlat[rhsIdx3+3] +
-						lhsFlat[lhsIdx+4]*rhsFlat[rhsIdx3+4] +
-						lhsFlat[lhsIdx+5]*rhsFlat[rhsIdx3+5] +
-						lhsFlat[lhsIdx+6]*rhsFlat[rhsIdx3+6] +
-						lhsFlat[lhsIdx+7]*rhsFlat[rhsIdx3+7]
-					lhsIdx += 8
-					rhsIdx += 8
+				var sum0, sum1, sum2, sum3 T
+
+				// SIMD acceleration for float32 on ARM64 using NEON Group4.
```

Comment:
So, the dotGeneralKernel is the very inner loop, and it would be nice to avoid the "if-clause" inside it.

What about registering the NEON version of buildDotGeneralKernel with `dotGeneralKernelDTypeMap` for `float32` in NEON, as opposed to the "if-clause" in the middle of the row loop ?

Also, because maybe NEON have different optimal parameters of how many loops to unroll (here I do with 4 rows at a time).